modules/images/regenerate-existing-images/background-process/class-perflab-background-job.php

```diff
@@ -0,0 +1,454 @@
+
+		return false;
+	}
+
+	/**
+	 * Retrieves the job status from its meta information.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string
+	 */
+	public function get_status() {
+		return (string) get_term_meta( $this->job_id, self::JOB_STATUS_META_KEY, true );
+	}
+
+	/**
+	 * Retrieves the job data.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array|null
+	 */
+	public function data() {
+		/**
+		 * If we have cached data, return it.
+		 */
+		if ( ! is_null( $this->data ) ) {
+			return $this->data;
+		}
+
+		$this->data = get_term_meta( $this->job_id, self::JOB_DATA_META_KEY, true );
+
+		return $this->data;
+	}
+
+	/**
+	 * Creates a new job in the queue.
+	 *
+	 * Job refers to the term and queue refers to the `background_job` taxonomy.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $name Name of job identifier.
+	 * @param array  $data Data for the job.
+	 *
+	 * @return int|WP_Error
+	 */
+	public function create( $name, array $data ) {
+		// @todo Replace taxonomy name with constant.
+		// Create job only when job ID for current instance is non-zero and term doesn't exist.
+		if ( $this->job_id > 0 && term_exists( $this->job_id, 'background_job' ) ) {
+			return new WP_Error( __( 'Cannot create the job as it exists already.', 'performance-lab' ) );
+		}
+
+		$term_name = 'job_' . microtime();
+
+		// Insert the new job in queue.
+		$term_data = wp_insert_term(
+			$term_name,
+			'background_job',
+			array(
+				'slug' => $term_name,
+			)
+		);
+
+		if ( ! is_wp_error( $term_data ) ) {
+			// Set object properties for instance as soon as job is created.
+			$this->job_id = $term_data['term_id'];
+			$this->name   = sanitize_text_field( $name );
+			$this->data   = $data;
+
+			// Set the queued status for freshly created jobs.
+			$this->set_status( self::JOB_STATUS_QUEUED );
+
+			update_term_meta( $this->job_id, self::JOB_DATA_META_KEY, $this->data );
+			update_term_meta( $this->job_id, self::JOB_NAME_META_KEY, $this->name );
+
+			/**
+			 * Fires when the job has been created successfully.
+			 *
+			 * @since n.e.x.t
+			 *
+			 * @param int    $job_id Job ID.
+			 * @param string $name Job name.
+			 * @param array  $data Job data.
+			 */
+			do_action( 'perflab_job_created', $this->job_id, $this->name, $this->data );
+		}
+
+		return $term_data;
+	}
+
+	/**
+	 * Get the batch for current run.
+	 *
+	 * This will return the items to process in the current batch.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
+	 */
+	public function batch() {
+		$items = array();
+
+		/**
+		 * Filters the items for the current batch of job.
+		 *
+		 * By default this will be empty array. Consumer code will need to use this filter in
+		 * order to return the list of items which needs to be processed in current batch.
+		 *
+		 * @param array  $items  Items for the current batch.
+		 * @param int    $job_id Background job ID.
+		 * @param string $name   Job identifier or name.
+		 * @param array  $data   Job data.
+		 *
+		 * @since n.e.x.t
```

Comment:
`@since` should come before `@param`
```suggestion
		 * @since n.e.x.t
		 *
		 * @param array  $items  Items for the current batch.
		 * @param int    $job_id Background job ID.
		 * @param string $name   Job identifier or name.
		 * @param array  $data   Job data.
```